File: spellbound-ngram/spellbound-ngram-scala/src/test/scala/net/xrrocha/scala/spellbound/ngram/SettingsTest.scala
package net.xrrocha.scala.spellbound.ngram

import com.typesafe.config.Config
import com.typesafe.scalalogging.LazyLogging
import org.scalatest.FunSuite

class SettingsTest extends FunSuite with LazyLogging {

  val config: Config = new GlobalConfigSettings {}.globalConfig

  test("Parses application.conf correctly and completely") {
    assert(config.getInt("spellingResources.ngramLength") == 3)
    assert(config.getString("spellingResources.dictionaryLocation") == "data/dictionary.tsv")
    assert(config.getString("spellingResources.ngramLocation") == "data/ngrams.tsv")
    assert(config.getDouble("stringMetric.maxDistance") == .275)
    assert(config.getString("stringMetric.stringMetric") == "damerau")
  }
}

File: spellbound-ngram/spellbound-ngram-kotlin/src/test/kotlin/net/xrrocha/kotlin/spellbound/ngram/SettingsTest.kt
package net.xrrocha.kotlin.spellbound.ngram

import net.xrrocha.kotlin.spellbound.ngram.Settings.config
import org.junit.Test
import kotlin.test.assertEquals

class SettingsTest {

  @Test
  fun parsesApplicationConf() {
    assertEquals("../data/ngrams.tsv", config.getString("files.ngramFilename"))
    assertEquals("../data/dictionary.tsv", config.getString("files.dictionaryFilename"))
    assertEquals(3, config.getInt("parameters.ngramLength"))
    assertEquals(.75, config.getDouble("parameters.maxDistance"))
    assertEquals("levenshtein", config.getString("parameters.stringMetric"))
  }
}